contracts/utils/FunderV2.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "./SelfMulticall.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/Create2.sol";
+import "./FunderDepositoryV2.sol";
+
+contract FunderV2 is SelfMulticall {
+    mapping(address => mapping(bytes32 => address payable))
+        public ownerToRootToFunderDepositoryAddress;
+
+    receive() external payable {}
+
+    function deployFunderDepository(
+        address owner,
+        bytes32 root
+    ) external returns (address payable funderDepository) {
+        // Owner allowed to be zero
+        require(root != bytes32(0), "Root zero");
+        funderDepository = payable(
+            new FunderDepositoryV2{salt: bytes32(0)}(owner, root)
+        );
+        ownerToRootToFunderDepositoryAddress[owner][root] = funderDepository;
+        // Emit event
+    }
+
+    function withdraw(
+        address owner,
```

Comment:
This argument seems redundant since we are checking that is equal to msg.sender, right?

Reply:
Good point